gamedata/scripts/LuaPanda.lua

```diff
@@ -0,0 +1,3628 @@
+    currentRunState == runState.STEPOUT_STOP or
+    currentRunState == runState.HIT_BREAKPOINT or
+    currentRunState == runState.STOP_ON_ENTRY
+    then
+        this.sendLuaMemory();
+        this.receiveMessage(MAX_TIMEOUT_SEC);
+        return
+    end
+end
+
+-----------------------------------------------------------------------------
+-- 调试器核心方法
+-----------------------------------------------------------------------------
+
+------------------------堆栈管理-------------------------
+
+
+--getStackTable需要建立stackTable，保存每层的lua函数实例(用来取upvalue)，保存函数展示层级和ly的关系(便于根据前端传来的stackId查局部变量)
+-- @level 要获取的层级
+function this.getStackTable( level )
+    local functionLevel = 0
+    if hookLib ~= nil then
+        functionLevel = level or HOOK_LEVEL;
+    else
+        functionLevel = level or this.getSpecificFunctionStackLevel(lastRunFunction.func);
+    end
+    local stackTab = {};
+    local userFuncSteakLevel = 0; --用户函数的steaklevel
+    local clevel = 0
+    repeat
+        local info = debug.getinfo(functionLevel, "SlLnf")
+        if info == nil then
+            break;
+        end
+        if info.source ~= "=[C]" then
+            local ss = {};
+            ss.file = this.getPath(info);
+            local oPathFormated = this.formatOpath(info.source) ; --从lua虚拟机获得的原始路径, 它用于帮助定位VScode端原始lua文件的位置(存在重名文件的情况)。
+            ss.oPath = this.truncatedPath(oPathFormated, truncatedOPath);
+            ss.name = "文件名"; --这里要做截取
+            ss.line = tostring(info.currentline);
+            --使用hookLib时，堆栈有偏移量，这里统一调用栈顶编号2
+            local ssindex = functionLevel - 3;
+            if hookLib ~= nil then
+                ssindex = ssindex + 2;
+            end
+            ss.index = tostring(ssindex);
+            table.insert(stackTab,ss);
+            --把数据存入currentCallStack
+            local callStackInfo = {};
+            callStackInfo.name = ss.file;
+            callStackInfo.line = ss.line;
+            callStackInfo.func = info.func;                     --保存的function
+            callStackInfo.realLy = functionLevel;               --真实堆栈层functionLevel(仅debug时用)
+            table.insert(currentCallStack, callStackInfo);
+
+            --level赋值
+            if userFuncSteakLevel == 0 then
+                userFuncSteakLevel = functionLevel;
+            end
+        else
+            local callStackInfo = {};
+            callStackInfo.name = info.source;
+            callStackInfo.line = info.currentline;              --C函数行号
+            callStackInfo.func = info.func;                     --保存的function
+            callStackInfo.realLy = functionLevel;               --真实堆栈层functionLevel(仅debug时用)
+            table.insert(currentCallStack, callStackInfo);
+            clevel = clevel + 1
+        end
+        functionLevel = functionLevel + 1;
+    until info == nil
+    return stackTab, userFuncSteakLevel;
+end
+
+-- 把路径中去除后缀部分的.变为/, 
+-- @filePath 被替换的路径
+-- @ext      后缀(后缀前的 . 不会被替换)
+function this.changePotToSep(filePath, ext)
```

Comment:
@ForserX , the core issue was here: the function incorrectly converted `test.script` into `test/script` and ultimately transformed it into `test/script.script`.

As you can see from the function's documentation, lmao, for the ext argument, it explicitly states that the dot before extensions shouldn't be converted to a slash. However, they never implemented this logic in the code. I had to add this functionality myself, and now `test.script` is correctly identified as `test.script`.